server/app/repository/ApplicationRepository.java

```diff
@@ -147,7 +148,10 @@ private ApplicationModel submitApplicationInternal(
         app.setLifecycleStage(LifecycleStage.OBSOLETE);
         app.save();
       }
-
+      application.setEligibilityDetermination(
```

Comment:
question (maybe blocking): Something about this seems off -- I need to think about it a little more, but I think this is where you're handling the case where `eligibilityDetermination` is null, which I'm pretty sure will never be the case. So then you're setting the eligibility to the existing eligibility on the application, which might be circular logic. Feel free to let me know if I'm not thinking about this correctly, otherwise I will look into in more depth tomorrow!

Reply:
Agreed. Is this defaulting it to NOT_COMPUTED in the case where it's unset? But I don't think it's ever not set, so I don't think we need this.